presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkOutputOperator.java

```diff
@@ -259,13 +260,19 @@ public void addInput(Page page)
             byte[] rowBytes = output.size() == 0 ? new byte[0] : output.getUnderlyingSlice().byteArray();
             if (shouldReplicate) {
                 for (int i = 0; i < partitionFunction.getPartitionCount(); i++) {
-                    appendRow(new PrestoSparkMutableRow(i, output.size(), rowBytes));
+                    PrestoSparkMutableRow row = new PrestoSparkMutableRow();
```

Comment:
At this moment you we still allocate one `PrestoSparkMutableRow` per row. Is this going to be addressed in later commit? (or in the future? :)

Reply:
Yeah. This  will be addressed in the next two commits.